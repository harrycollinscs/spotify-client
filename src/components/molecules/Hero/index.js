import React from 'react';
import styled from 'styled-components';

const HeroContainer = styled.div`
  background-image: linear-gradient(#3649B4, #AC2F94);
  height: 40vh;
  width: 100%;
  padding: 100px;
`;

const StyledTitle = styled.h1`
  font-size: 50px;
  font-weight: bold;
  color: white;
`;

const StyledContent = styled.p`
  color: white;
`;

const Hero = ({ title = "", content = "" }) => (
  <HeroContainer>
    <StyledTitle>{title}</StyledTitle>
    <StyledContent>{content}</StyledContent>
  </HeroContainer>
);

export default Hero;