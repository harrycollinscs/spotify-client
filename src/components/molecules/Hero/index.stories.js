import React from "react";
import Hero from "./index";

export default {
  title: "Molecules/Hero",
  component: Hero,
};

const Template = (args) => <Hero {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: "",
  content: "",
  withBackButton: false,
};

export const WithTitle = Template.bind({});
WithTitle.args = {
  title: "Dummy title",
  content: "",
  withBackButton: false,
};

export const WithContent = Template.bind({});
WithContent.args = {
  title: "",
  content: "Here's some dummy content for the Hero component",
  withBackButton: false,
};

export const WithBackButton = Template.bind({});
WithBackButton.args = {
  title: "",
  content: "",
  withBackButton: true,
};
