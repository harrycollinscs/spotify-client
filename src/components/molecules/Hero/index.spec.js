import React from "react";
import Hero from "./index";
import { shallow } from "enzyme";

describe("Molecules/Hero", () => {
  const component = <Hero />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
