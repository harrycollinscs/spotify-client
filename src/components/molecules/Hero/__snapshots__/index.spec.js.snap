// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Hero Should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      withBack={false}
    >
      <styled.h1 />
      <styled.p />
    </styled.div>
  </styled.div>
</styled.div>
`;
